### PR TITLE
Tracking refactors, Debug tracking, and more swift 3

### PR DESCRIPTION
**Pull request by colinta**
Went over all the tracking changes w/ Casey, and this is the result!

- lots of shorter "screen viewed" tracking names, e.g. "Streams" instead of "StreamContainerViewController".
- tracking *more* screen events, in particular `viewDidAppear` now tracks as a "fresh sceen view"
- added deletes tracking (comments/posts), fixed double-tracking of loves
- refactored search tracking

The `DebugTrackingController` can be used to show tracking events in real time, Debug builds only.

Unrelated to tracking:

- swift3-ish changes in `CGRect`
- I removed `Crashlytics` and `Tracker.` calls from `Models/`
- the status bar wasn't showing/hiding due to (I think) a plist setting
- removed redundant `.slide` animation value (it was always `.slide`), and renamed notification to include `ShouldHide` intention language.
- removed the `FIXME:` nil comparison operators

### CODE REVIEW

File: Sources/Controllers/Omnibar/OmnibarViewController.swift [hunk 1]
```diff
@@ -8,14 +8,27 @@ import PINRemoteImage
 
 
 class OmnibarViewController: BaseElloViewController {
-    var keyboardWillShowObserver: NotificationObserver?
-    var keyboardWillHideObserver: NotificationObserver?
+    override func trackerName() -> String? { return "Omnibar" }
+    override func trackerProps() -> [String: AnyObject]? {
+        if let parentPost = parentPost {
```
Comment (steam) [inline Sources/Controllers/Omnibar/OmnibarViewController.swift]: If these let values are not used they can be nil checks i.e. if `parentPost != nil { ... }`

File: Sources/Controllers/Stream/CellDequeing/ProfileAvatarPresenter.swift [hunk 2]
```diff
@@ -45,6 +21,6 @@ struct ProfileAvatarPresenter {
             view.avatarURL = url
         }
 
-        view.badgeVisible = user.totalViewsCount == nil && user.categories?.count > 0
+        view.badgeVisible = user.totalViewsCount == nil && (user.categories?.count ?? 0) > 0
```
Comment (steam) [inline Sources/Controllers/Stream/CellDequeing/ProfileAvatarPresenter.swift]: What was the motivation for all these `?? 0` additions?

File: Sources/Extensions/CGRectExtensions.swift [hunk 3]
```diff
@@ -48,37 +48,37 @@ extension CGRect {
     }
 
 // MARK: dimension setters
-    func atOrigin(_ amt: CGPoint) -> CGRect {
+    func at(origin amt: CGPoint) -> CGRect {
```
Comment (steam) [inline Sources/Extensions/CGRectExtensions.swift]: Yeah Swift 3!

File: ShareExtension/ShareViewController.swift [hunk 4]
```diff
@@ -82,12 +82,12 @@ private extension ShareViewController {
             content: content,
             buyButtonURL: nil,
             success: { post in
-//                Tracker.sharedTracker.shareSuccessful()
+//                Tracker.shared.shareSuccessful()
```
Comment (colinta) [inline ShareExtension/ShareViewController.swift]: Huh, interesting - no, no clue.  I'll uncomment them, or make sure they're tracked somewhere.

File: Sources/Controllers/Omnibar/OmnibarViewController.swift (hunk 1 above)
Comment (colinta) [inline Sources/Controllers/Omnibar/OmnibarViewController.swift]: 👍

File: Sources/Controllers/Stream/CellDequeing/ProfileAvatarPresenter.swift (hunk 2 above)
Comment (colinta) [inline Sources/Controllers/Stream/CellDequeing/ProfileAvatarPresenter.swift]: To remove the `// FIXME` code - you can no longer compare `Optional<Int> > 0`

Comment (steam) [inline Sources/Controllers/Stream/CellDequeing/ProfileAvatarPresenter.swift]: Ah, yep. Makes sense.

File: ShareExtension/ShareViewController.swift (hunk 4 above)
Comment (colinta) [inline ShareExtension/ShareViewController.swift]: Ah, probably to minimize the ShareExtension target's dependencies.
```
commit c4f4bec01fcfd9b498a8b172cd9a587615cefbac
Author: Sean Dougherty <seancdougherty@gmail.com>
Date:   Tue Feb 16 17:40:54 2016 -0700

    remove references to Tracker/Analytics
```

File: Sources/Controllers/Omnibar/OmnibarViewController.swift (hunk 1 above)
Comment (colinta) [inline Sources/Controllers/Omnibar/OmnibarViewController.swift]: Oops, I forgot to push this commit before merging - but I had it locally, so it's merged into `swift3.0` now.

**Pull request merged**